src/repository.ts

```diff
@@ -97,20 +124,45 @@ export class StorageRepository implements Repository {
     return undefined;
   }
 
+  async findFlagsBySegment(segment: string): Promise<string[]> {
+    const result = [];
+    let keys = this.cache.keys();
+    if (this.store) {
+      keys = await this.store.keys();
+    }
+    for (const key of keys) {
+      const flag = await this.getFlag(key);
+      if (!flag) {
```

Comment:
So if any key doesn't match a flag, we return a blank array? Curious if it would make sense to `continue` instead?